### ISSUE TITLE
 Inability to Search and Display Performance Custom Performance Measurements with Numbers

### ISSUE THREAD

**Issue by Kobby-Bawuah** [getsentry/sentry#52758]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

Currently, I have set several measurements using the transaction.setMeasurement() method, which includes numeric values. For example:

```
transaction1.setMeasurement('frame.rate.avg', 50);
transaction1.setMeasurement('frame.rate.p50', 60);
transaction1.setMeasurement('frame.rate.a50', 70);
transaction1.setMeasurement('frame.rate.50', 80);
transaction1.setMeasurement('frame.rate.p1234', 10);
transaction1.setMeasurement('frame.rate.sdbibdijd', 20);
```

It show up in Sentry: 

![Screenshot 2023-07-12 at 5 26 35 PM](https://github.com/getsentry/sentry/assets/89524511/d28f1064-b596-415d-954c-b4bb4671cae7)

However, when attempting to search for these metrics in both the Discover and Dashboards features, it seems that only metrics without numbers are searchable. 

![Screenshot 2023-07-11 at 3 05 51 PM](https://github.com/getsentry/sentry/assets/89524511/cb18b0d9-a14e-4f06-854d-b2de5dc5449c)
![Screenshot 2023-07-12 at 5 28 10 PM](https://github.com/getsentry/sentry/assets/89524511/346a1ba3-e499-494f-a487-f17a72b3823b)

Additionally, these metrics are not available in the Discover columns unless the URL is manually edited to include them.
![Screenshot 2023-07-12 at 5 28 10 PM](https://github.com/getsentry/sentry/assets/89524511/27b9613f-adbd-47da-8e4f-9817d54e4440)

In summary: 
- Set measurements using the transaction.setMeasurement() method with numeric values.
- Attempt to search for these metrics using the Discover or Dashboards features.
- Observe that the metrics with numbers are not searchable and not available in the Discover columns by default.

### Expected Result

Metrics with numbers set using the transaction.setMeasurement() method should be searchable and visible in the Discover and Dashboards features without the need for manual URL modification.

### Actual Result

Metrics with numbers in their names are not searchable in both the Discover and Dashboards features. Additionally, these metrics are not available in the Discover columns. 

### Product Area

Dashboards

**Comment by xiaoshuangLi** [comment]
Is there any solutions ?? this is exactly my problems.

**Comment by Moreno-Gentili** [comment]
Any update about this? In my case `a.b` is showing up while `a.b.c` does not. Not sure if the amount of dots has anything to do with this issue.

**Comment by wmak** [comment]
Hey sorry this one fell off our Radar, I'm going to start some investigation into this and will have an update soon

**Comment by wmak** [comment]
The cases in the original issue are fixed by https://github.com/getsentry/sentry/pull/62580
but @Moreno-Gentili I wasn't able to reproduce any issues with `a.b.c` once 62580 merges if you still see issues with your measurements not showing up please feel free to reopen this :+1:
